### DISCUSSION TITLE
Issues with API

### DISCUSSION THREAD

**Discussion by Axon07** [magicbug/Cloudlog#861] [Q&A]
I have installed Cloudlog on a webserver hosted in the cloud but been having lots of issues getting the API to work. 
Had to wipe my DB as I initially set it up with HTTP and I was reading to use HTTPS for API and of course it's more secure. After wiping the DB and reinstalling with HTTPS I was still having issues with all the add-ons to work with Cloudlog. Initially I thought it was the add-on that was the issue but have tried multiple and none worked so I think it is the install of Cloudlog or provider missing something and not the add-on.

The addins I have tried are:
- CloudlogCAT (Windows)
- CloudlogCATQt (Windows)
- SatPC32 (Windows)
- CloudLog Offline (Android)

The provider does have the following PHP modules installed on the server.

bcmath | hash | mime_magic | Reflection | wddx
calendar | iconv | mssql | session | xml
ctype | imagick* | mysql | SimpleXML | xmlreader
curl | imap | mysqli | soap | xmlrpc
date | ionCube Loader | openssl | sockets | xmlwrite
dom | json | pcre | SourceGuardian | xsl
exif* | libxml | PDO | SPL | zip
filter | magickwand* | pdo_mysql | SQLite | zlib
ftp | mbstring | pdo_sqlite | standard
gd | mcrypt | posix | tidy
gettext | mhash | pspell | tokenizer

Some help would be greatly appreciated.

**Comment by magicbug** [comment]
If your using the latest branch I cant see why its not working, I'm guessing using a read-write key, whats PHP erroring regarding? anything in the server logs?

**Reply by Axon07** [reply]
Ok spent hours troubleshooting but figured it out.
I am using a Hostgator webhosting as a provider and they do not have OAuth PHP package installed on the server. I spun up a VPS and installed it on there and it worked with OAuth installed...
https://www.hostgator.com/help/article/php-modules
The Hostgator admin staff are reviewing it and may add the PHP module but as of right now nothing wrong with CloudLog.

Thanks for the quick support!
